vessim/data.py

```diff
@@ -0,0 +1,169 @@
+                    not needed if use_forecast is set to False.
+                `fill_method`: The fill_method of the TimeSeriesApi. If not specified,
+                    `bfill` is used.
+                `static_forecast`: Bool indicating if the forecast is static. If set
+                    to True, the forecast does not contain a `Request Timestamp`, but
+                    if not specified, the forecast is treated as non-static forecast.
+                    This is not needed if use_forecast is set to False.
+                `url`: String with a URL to a zip-file if data not locally available.
+        data_dir: Absolute path to the directory where the data is/should be located.
+        scale: Multiplies all data points with a value. Defaults to 1.0.
+        start_time: Shifts the data so that it starts at this timestamp if specified.
+            Defaults to None.
+        use_forecast: Bool indicating if forecast should be loaded. Default is true.
+
+    Returns:
+        The dataframe of actual data, the optional dataframe of forecast data and the
+        fill_method to be fed into a TimeSeriesApi.
+
+    Raises:
+        RuntimeError if dataset can not be loaded.
+    """
+    if isinstance(dataset, str):
+        dataset_dict: Dict = Datasets[dataset]
+    else:
+        dataset_dict = dataset
+
+    required_files = [dataset_dict["actual"]]
+    if use_forecast:
+        required_files.append(dataset_dict["forecast"])
+
+    dir_path = Path(data_dir or "").expanduser().resolve()
+
+    if not _check_files(required_files, dir_path):
+        if "url" not in dataset_dict.keys():
+            raise RuntimeError("Data files could not be found.")
+
+        print("Required data files not present. Try downloading...")
+        os.makedirs(dir_path, exist_ok=True)
+
+        try:
+            urllib.request.urlretrieve(dataset_dict["url"], dir_path / "dataset.zip")
+        except Exception:
+            raise RuntimeError(
+                f"Dataset could not be retrieved from url: {dataset_dict['url']}"
+            )
+
+        with ZipFile(dir_path / "dataset.zip", "r") as zip_ref:
+            zip_ref.extractall(path=dir_path)
+        os.remove(dir_path / "dataset.zip")
+        print("Successfully downloaded and unpacked data files.")
+
+    actual = _read_data_from_csv(
+        dir_path / dataset_dict["actual"], index_cols=[0], scale=scale
+    )
+
+    forecast: Optional[PandasObject] = None
+    if use_forecast:
+        if dataset_dict.get("static_forecast", False):
+            # There is only one timestamp present in the forecast (static forecast)
+            index_cols: List[int] = [0]
+        else:
+            # There are two timestamps present in the forecast (non-static forecast)
+            index_cols = [0, 1]
+
+        forecast = _read_data_from_csv(
+            dir_path / dataset_dict["forecast"], index_cols=index_cols, scale=scale
+        )
+
+    if start_time is not None:
```

Comment:
maybe we want to print some logs here that the loaded data has been shifted by X